src/main/java/com/puppycrawl/tools/checkstyle/Checker.java

```diff
@@ -66,7 +66,7 @@
     public static final String EXCEPTION_MSG = "general.exception";
 
     /** Logger for Checker. */
-    private static final Log LOG = LogFactory.getLog(Checker.class);
+    private final Log log;
```

Comment:
Updating the main class' logger variable has these difficulties.
1. All the methods in the main that assess the log variable are defined as static
2. The main class cannot be instantiated to call the non-static log in the methods that call the log.

Any thought on what to do?